File: swagger/models/checkout_item_wrapper.py
from .base_model import BaseModel

from .checkout_item import CheckoutItem

class CheckoutItemWrapper(BaseModel):
	"""
	no description available

	Attributes:

		item
		Information about an item in the cart.

		discount_amount
		The amount the item is discounted. This parameter is ignored for
		packages.

		appointment_booking_requests
		A list of appointments to be booked then paid for by this item. This
		parameter applies only to pricing option items.

		enrollment_ids
		A list of enrollment IDs that this item is to pay for. This parameter
		applies only to pricing option items.

		class_ids
		A list of class IDs that this item is to pay for. This parameter
		applies only to pricing option items.

		course_ids
		A list of course IDs that this item is to pay for. This parameter
		applies only to pricing option items.

		visit_ids
		A list of visit IDs that this item is to pay for. This parameter
		applies only to pricing option items.

		appointment_ids
		A list of appointment IDs that this item is to reconcile.

		id
		The item’s unique ID within the cart.

		quantity
		The number of this item to be purchased.
	"""

	input_map = {
		'Item': ('item', Item),
		'DiscountAmount': 'discount_amount',
		'AppointmentBookingRequests': 'appointment_booking_requests',
		'EnrollmentIds': 'enrollment_ids',
		'ClassIds': 'class_ids',
		'CourseIds': 'course_ids',
		'VisitIds': 'visit_ids',
		'AppointmentIds': 'appointment_ids',
		'Id': 'id',
		'Quantity': 'quantity',
		}

	output_map = {
		'item': 'Item',
		'discount_amount': 'DiscountAmount',
		'appointment_booking_requests': 'AppointmentBookingRequests',
		'enrollment_ids': 'EnrollmentIds',
		'class_ids': 'ClassIds',
		'course_ids': 'CourseIds',
		'visit_ids': 'VisitIds',
		'appointment_ids': 'AppointmentIds',
		'id': 'Id',
		'quantity': 'Quantity',
		}

		'Item': ('item', Item),
	discount_amount = None
	appointment_booking_requests = None
	enrollment_ids = None
	class_ids = None
	course_ids = None
	visit_ids = None
	appointment_ids = None
	id = None
	quantity = None

File: swagger/Models/CheckoutItemWrapper.php
<?php

namespace App\MindBody\Models;

/**
 * no description available
 */
class CheckoutItemWrapper extends BaseModel
{
	/**
	 * Information about an item in the cart.
	 * 
	 * @var CheckoutItem 
	 */
	public $item;
	/**
	 * The amount the item is discounted. This parameter is ignored for
	 * packages.
	 * 
	 * @var number format:double
	 */
	public $discountAmount;
	/**
	 * A list of appointments to be booked then paid for by this item. This
	 * parameter applies only to pricing option items.
	 * 
	 * @var CheckoutAppointmentBookingRequest[] 
	 */
	public $appointmentBookingRequests;
	/**
	 * A list of enrollment IDs that this item is to pay for. This parameter
	 * applies only to pricing option items.
	 * 
	 * @var integer[] 
	 */
	public $enrollmentIds;
	/**
	 * A list of class IDs that this item is to pay for. This parameter
	 * applies only to pricing option items.
	 * 
	 * @var integer[] 
	 */
	public $classIds;
	/**
	 * A list of course IDs that this item is to pay for. This parameter
	 * applies only to pricing option items.
	 * 
	 * @var integer[] 
	 */
	public $courseIds;
	/**
	 * A list of visit IDs that this item is to pay for. This parameter
	 * applies only to pricing option items.
	 * 
	 * @var integer[] 
	 */
	public $visitIds;
	/**
	 * A list of appointment IDs that this item is to reconcile.
	 * 
	 * @var integer[] 
	 */
	public $appointmentIds;
	/**
	 * The item’s unique ID within the cart.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * The number of this item to be purchased.
	 * 
	 * @var integer format:int32
	 */
	public $quantity;

	protected function getInputMap()
	{
		return [
			'Item' => ['item', CheckoutItem::class, null],
			'DiscountAmount' => ['discountAmount', 'number', 'double'],
			'AppointmentBookingRequests' => ['appointmentBookingRequests', 'array', CheckoutAppointmentBookingRequest::class],
			'EnrollmentIds' => ['enrollmentIds', 'array', null],
			'ClassIds' => ['classIds', 'array', null],
			'CourseIds' => ['courseIds', 'array', null],
			'VisitIds' => ['visitIds', 'array', null],
			'AppointmentIds' => ['appointmentIds', 'array', null],
			'Id' => ['id', 'integer', 'int32'],
			'Quantity' => ['quantity', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'item' => ['Item', CheckoutItem::class, null],
			'discountAmount' => ['DiscountAmount', 'number', 'double'],
			'appointmentBookingRequests' => ['AppointmentBookingRequests', 'array', CheckoutAppointmentBookingRequest::class],
			'enrollmentIds' => ['EnrollmentIds', 'array', null],
			'classIds' => ['ClassIds', 'array', null],
			'courseIds' => ['CourseIds', 'array', null],
			'visitIds' => ['VisitIds', 'array', null],
			'appointmentIds' => ['AppointmentIds', 'array', null],
			'id' => ['Id', 'integer', 'int32'],
			'quantity' => ['Quantity', 'integer', 'int32'],
			];
	}
}
